services/httpd/response_writer.go

```diff
@@ -98,13 +98,9 @@ func (w *bytesCountWriter) Write(data []byte) (int, error) {
 
 // WriteResponse writes the response using the formatter.
 func (w *responseWriter) WriteResponse(resp Response) (int, error) {
-	n := 0
 	writer := bytesCountWriter{w: w.ResponseWriter}
 	err := w.formatter.WriteResponse(&writer, resp)
-	if err != nil {
-		n, _ = WriteError(w, err)
```

Comment:
WriteError calls WriteResponse, and if the ResponseWriter is bad and returning errors, that calls back to WriteError. This was introduced recently, never released.